### PR TITLE
docs(datatype): allow markdown in @see links

### PR DESCRIPTION
**Pull request by matthewmayer**
The backtick in the `@see` causes the link to get broken and the second see also not to appear:

On https://next.fakerjs.dev/api/datatype.html#number

Screenshot 2022-12-16 at 14 15 45

after fix:

Screenshot 2022-12-16 at 14 22 50

### CODE REVIEW

**Comment by Shinigami92** [comment]
This is for rendering in VSCode tooltips



Therefore I assume instead of changing the source text, we need to somehow fix the docs automatisation text extractor

**Comment by matthewmayer** [comment]
i guess its this method? https://github.com/faker-js/faker/blob/next/scripts/apidoc/utils.ts#L114 but dont really understand what its doing or why there are two cases for a length of 1 or 3?

**Comment by matthewmayer** [comment]
previous PR where this was implemented: https://github.com/faker-js/faker/pull/1270/files

**Comment by matthewmayer** [comment]
I tried fixing the @see extraction, which works OK, although it renders like this in the docs, maybe would be better if the `` rendered as `<code></code>`

image

**Comment by ST-DDT** [comment]
You probably have to call mdToHtml somewhere.

File: scripts/apidoc/utils.ts [hunk 1]
```diff
@@ -117,9 +117,21 @@ export function extractSeeAlsos(signature?: SignatureReflection): string[] {
     if (content.length === 1) {
       return joinTagContent(tag);
     }
```
Comment (ST-DDT) [inline scripts/apidoc/utils.ts]: This is no longer needed.

File: scripts/apidoc/utils.ts [hunk 2]
```diff
@@ -117,9 +117,21 @@ export function extractSeeAlsos(signature?: SignatureReflection): string[] {
...
-    return tag.content
-      .filter((_, index) => index % 3 === 1) // ['-', 'content', '\n']
-      .map((part) => part.text);
+    // If the @see tag contains code in backticks, the content is split into multiple parts.
+    // So we join together, split on newlines and filter out empty tags.
+    const links = tag.content
+      .map((c) => c.text)
+      .join('')
```
Comment (ST-DDT) [inline scripts/apidoc/utils.ts]:
```suggestion
    return joinTagParts(tag.content)
```

File: scripts/apidoc/utils.ts [hunk 3]
```diff
@@ -117,9 +117,21 @@ export function extractSeeAlsos(signature?: SignatureReflection): string[] {
...
+      .split('\n')
+      .map((link) => {
+        link = link.trim();
+        if (link.startsWith('-')) {
+          link = link.slice(1).trim();
+        }
+        return link;
+      })
+      .filter((link) => link.length);
```
Comment (ST-DDT) [inline scripts/apidoc/utils.ts]:
```suggestion
      .filter((link) => link);
```

An empty string is also falsy. Alternatively, trim after `joinTagParts`.

File: scripts/apidoc/utils.ts [hunk 4]
```diff
@@ -117,9 +117,21 @@ export function extractSeeAlsos(signature?: SignatureReflection): string[] {
...
+    return links;
```
Comment (ST-DDT) [inline scripts/apidoc/utils.ts]:
```suggestion
```

File: scripts/apidoc/utils.ts [hunk 5]
```diff
@@ -117,9 +117,21 @@ export function extractSeeAlsos(signature?: SignatureReflection): string[] {
...
+    // If the @see tag contains code in backticks, the content is split into multiple parts.
+    // So we join together, split on newlines and filter out empty tags.
+    const links = tag.content
+      .map((c) => c.text)
+      .join('')
+      .split('\n')
+      .map((link) => {
+        link = link.trim();
+        if (link.startsWith('-')) {
+          link = link.slice(1).trim();
```
Comment (ST-DDT) [inline scripts/apidoc/utils.ts:123]: Why do you use `.slice(1).trim()` instead of `.slice(2)`? (Just curious)

**Review by ST-DDT — COMMENTED** [review]
Thanks for fixing this.
I only have some minor nitpicking, but other than that this is ready to be merged.

File: scripts/apidoc/utils.ts (hunk 5 above)
Comment (matthewmayer) [reply scripts/apidoc/utils.ts:123]: in case in theory it began with `-foo` not `- foo` (maybe not possible in reality)

**Review by ST-DDT — DISMISSED** [review]

**Review by xDivisionByZerox — DISMISSED** [review]

File: scripts/apidoc/signature.ts [hunk 6]
```diff
@@ -80,9 +80,14 @@ function comparableSanitizedHtml(html: string): string {
     .replace(/&quot;/g, '"')
     .replace(/&#39;/g, "'");
 }
-
-function mdToHtml(md: string): string {
-  const rawHtml = markdown.render(md);
+/**
```
Comment (Shinigami92) [inline scripts/apidoc/signature.ts:84]:
```suggestion

/**
```

File: scripts/apidoc/signature.ts [hunk 7]
```diff
@@ -80,9 +80,14 @@ function comparableSanitizedHtml(html: string): string {
...
+ * Converts Markdown to an HTML string and sanitizes it.
+ * @param md The markdown to convert.
+ * @param inline Whether to render the markdown as inline, without a wrapping <p> tag. Default false.
```
Comment (Shinigami92) [inline scripts/apidoc/signature.ts]:
```suggestion
 * @param inline Whether to render the markdown as inline, without a wrapping `<p>` tag. Defaults to `false`.
```

File: scripts/apidoc/signature.ts [hunk 8]
```diff
@@ -164,7 +169,9 @@ export function analyzeSignature(
     examples += `${exampleTags.join('\n').trim()}\n`;
   }
 
-  const seeAlsos = extractSeeAlsos(signature);
+  const seeAlsos = extractSeeAlsos(signature).map((seealso) =>
+    mdToHtml(seealso, true)
+  );
```
Comment (Shinigami92) [inline scripts/apidoc/signature.ts]:
```suggestion
  const seeAlsos = extractSeeAlsos(signature).map((seeAlso) =>
    mdToHtml(seeAlso, true)
  );
```

**Review by Shinigami92 — DISMISSED** [review]

**Review by ST-DDT — DISMISSED** [review]

**Comment by ST-DDT** [comment]
The snapshots need updating after the last typedoc/vitepress update

**Comment by ST-DDT** [comment]
The PR is still failing.

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**